server/server.go

```diff
@@ -543,143 +432,152 @@ func (s *server) CreateBlockchains(
 	s.clusterInfo.CustomChainsHealthy = false
 
-	// start non-blocking to install custom chains (if applicable)
-	// the user is expected to poll cluster status
-	readyCh := make(chan struct{})
-	go s.network.createBlockchains(ctx, chainSpecs, readyCh)
-
 	// update cluster info non-blocking
 	// the user is expected to poll this latest information
 	// to decide cluster/subnet readiness
 	go func() {
-		s.waitChAndUpdateClusterInfo("custom chains", readyCh, true)
+		ctx, cancel := context.WithTimeout(context.Background(), waitForHealthyTimeout)
+		defer cancel()
+		err := s.network.CreateChains(ctx, chainSpecs)
+		s.mu.Lock()
+		defer s.mu.Unlock()
+		if err != nil {
+			s.log.Error("failed to create blockchains", zap.Error(err))
+			s.stopAndRemoveNetwork()
+			return
+		} else {
+			s.updateClusterInfo()
+		}
+		s.log.Info("custom chains created")
 	}()
-
 	return &rpcpb.CreateBlockchainsResponse{ClusterInfo: s.clusterInfo}, nil
 }
 
-func (s *server) CreateSubnets(ctx context.Context, req *rpcpb.CreateSubnetsRequest) (*rpcpb.CreateSubnetsResponse, error) {
-	// if timeout is too small or not set, default to 5-min
-	if deadline, ok := ctx.Deadline(); !ok || time.Until(deadline) < defaultStartTimeout {
-		var cancel context.CancelFunc
-		ctx, cancel = context.WithTimeout(context.Background(), defaultStartTimeout)
-		_ = cancel // don't call since "start" is async, "curl" may not specify timeout
-		s.log.Info("received start request with default timeout", zap.String("timeout", defaultStartTimeout.String()))
-	} else {
-		s.log.Info("received start request with existing timeout", zap.String("timeout", deadline.String()))
+func (s *server) CreateSubnets(_ context.Context, req *rpcpb.CreateSubnetsRequest) (*rpcpb.CreateSubnetsResponse, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.network == nil {
+		return nil, ErrNetworkNotRunning
 	}
 
 	s.log.Debug("CreateSubnets", zap.Uint32("num-subnets", req.GetNumSubnets()))
 
-	if s.getNetwork() == nil {
-		return nil, ErrNotBootstrapped
-	}
-
 	// default behaviour without args is to create one subnet
 	numSubnets := req.GetNumSubnets()
 	if numSubnets == 0 {
 		numSubnets = 1
 	}
 
 	s.log.Info("waiting for local cluster readiness")
-	if err := s.getNetwork().waitForLocalClusterReady(ctx); err != nil {
-		return nil, err
-	}
-
-	s.mu.Lock()
-	defer s.mu.Unlock()
 
 	s.clusterInfo.Healthy = false
 	s.clusterInfo.CustomChainsHealthy = false
 
-	// start non-blocking to add subnets
-	// the user is expected to poll cluster status
-	readyCh := make(chan struct{})
-	go s.network.createSubnets(ctx, numSubnets, readyCh)
-
 	// update cluster info non-blocking
 	// the user is expected to poll this latest information
 	// to decide cluster/subnet readiness
 	go func() {
-		s.waitChAndUpdateClusterInfo("custom chains", readyCh, true)
+		ctx, cancel := context.WithTimeout(context.Background(), waitForHealthyTimeout)
+		defer cancel()
+		err := s.network.CreateSubnets(ctx, numSubnets)
```

Comment:
we want this error to be returned by WaitForHealthy

Reply:
WaitForHealthy not only waits for network to be on, it is also giving to the client the reason for the network failure if there was any. Usually that error was hidden in ANR logs, recently we added this function so users like CLI can check the error and provide it to the user.
All the async errors from go routines, due to calling CreateChains,  CreateSubnets, AwaitAndUpdateNetworkInfo, should be passed to this function using some mechanism.